vita3k/modules/SceKernelThreadMgr/SceThreadmgr.cpp

```diff
@@ -617,8 +658,17 @@ EXPORT(int, sceKernelCloseTimer) {
     return STUBBED("References not implemented.");
 }
 
-EXPORT(int, sceKernelCreateCallback) {
-    return UNIMPLEMENTED();
+EXPORT(SceUID, sceKernelCreateCallback, Ptr<char> name, SceUInt32 attr, Ptr<SceKernelCallbackFunction> callbackFunc, Ptr<void> pCommon) {
+    if (attr || !callbackFunc.address())
+        return RET_ERROR(SCE_KERNEL_ERROR_ILLEGAL_ATTR);
+
+    ThreadStatePtr thread = host.kernel.get_thread(thread_id);
+    std::string cb_name = name.get(host.mem);
+    auto cb = std::make_shared<Callback>(thread_id, thread, cb_name, callbackFunc, pCommon);
+    SceUID cb_uid = host.kernel.get_next_uid();
```

Comment:
You need lock here and any other places that's touching callbacks to prevent race condition. Use host.kernel.mutex.

Reply:
select all line where you want it, i put it and give one exemple